src/parser/types.rs

```diff
@@ -1258,9 +1264,53 @@ pub enum StmtKind {
         range: Range,
         body: Vec<Stmt>,
     },
+
+    // `for item in vec { <body> }`
+    IteratorLoop {
+        var: Ident,
+        iterator: Box<Expr>,
+        body: Vec<Stmt>,
+    },
```

Comment:
late to the party, but wouldn't it be cleaner (and perhaps safer since forloops have annoying logic for generics) to have a common structure for them?

```rust
ForLoop {
  var: Ident,
  range: RangeOrIterator,
  body: Vec<Stmt>,
}
```

Reply:
cf https://github.com/zksecurity/noname/issues/177